### PR TITLE
11 22 fixing

### PR DESCRIPTION
**Pull request by Lilyhanig**
- fixes doctoring spacing
- adds a check sum to verify results weren't tampered with
- changes to chronology in aggregation
- tests for chronology
- tests for reading and writing results

### CODE REVIEW

File: test/test_load_simulation.jl [hunk 1]
```diff
@@ -0,0 +1,91 @@
+include("../Simulations_preload.jl");
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: Just include the code here from Simulation preload that is used

File: test/test_load_simulation.jl [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                                    :ILoads =>  DeviceModel(PSY.InterruptibleLoad, PSI.DispatchablePowerLoad),
+                                    )       
+template_ed= OperationsTemplate(CopperPlatePowerModel, devices, branches, services);
+
+GLPK_optimizer = with_optimizer(GLPK.Optimizer)
+
+stages = Dict(1 => Stage(template_uc, 24, Hour(24), 1, c_sys5_uc, GLPK_optimizer,  Dict(0 => Sequential())),
+              2 => Stage(template_ed, 12, Minute(5), 24, c_sys5_ed, GLPK_optimizer, Dict(1 => Synchronize(24,1), 0 => Sequential()), TimeStatusChange(:ON_ThermalStandard)))
+
+sim = Simulation("test", 2, stages, "/Users/lhanig/Documents/"; verbose = true)
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: You can't specify a directory that is local to you, try something inside the package like this 
`test_dir = joinpath(base_dir,"test/results"); mkdir(test_dir)`

File: test/test_load_simulation.jl [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+sim_results, date_run = execute!(sim)
+step = ["step-1", "step-2"]
+stages = ["stage-1", "stage-2"]
+
+@testset "testing reading and writing to the results folder" begin
+    for (ix, s) in enumerate(stages)
+        files = collect(readdir("/Users/lhanig/Documents/test/$date_run/results"))
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: Again a local directory

File: test/test_load_simulation.jl [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        for f in files
+            rm("/Users/lhanig/Documents/test/$date_run/results/$f")
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: local directory

File: test/test_load_simulation.jl [hunk 5]
```diff
@@ -1,4 +1,105 @@
...
+using Test
+const PJ = ParameterJuMP
+import PowerModels
+const PM = PowerModels
+ipopt_optimizer = with_optimizer(Ipopt.Optimizer, print_level = 4)
+GLPK_optimizer = with_optimizer(GLPK.Optimizer, msg_lev = GLPK.MSG_ALL)
+using PowerSimulations
+const PSI = PowerSimulations
+using Random
+Random.seed!(232)
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: All these include are made in `runtests.jl` you dont need to add them here, please remove these

File: test/test_chronology.jl [hunk 6]
```diff
@@ -1,5 +1,105 @@
...
+using Test
+const PJ = ParameterJuMP
+import PowerModels
+const PM = PowerModels
+ipopt_optimizer = with_optimizer(Ipopt.Optimizer, print_level = 4)
+GLPK_optimizer = with_optimizer(GLPK.Optimizer, msg_lev = GLPK.MSG_ALL)
+using PowerSimulations
+const PSI = PowerSimulations
+using Random
+Random.seed!(232)
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: Please remove the `using/import `

File: test/test_chronology.jl [hunk 7]
```diff
@@ -1,5 +1,105 @@
...
+    end
+    for (ix, r) in enumerate(get_components(RenewableGen, c_sys5_uc))
+        add_forecast!(c_sys5_uc, r, Deterministic("get_rating", ren_timeseries_DA[t][ix]))
+    end
+    for (ix, i) in enumerate(get_components(InterruptibleLoad, c_sys5_uc))
+        add_forecast!(c_sys5_uc, i, Deterministic("get_maxactivepower", Iload_timeseries_DA[t][ix]))
+    end
+end
+using TimeSeries
+using Dates
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: Again if you need to import a package do it in the `runtests.jl` file

File: test/test_load_simulation.jl [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+end
+
+@testset "testing load simulation results functionality stage 2" begin
+    for (ix,s) in enumerate(stages)
+        variable = (collect(keys(sim.stages[ix].canonical.variables)))
+        results = load_simulation_results(s, sim_results)
+        res = load_simulation_results(s, step, variable, sim_results)
+        @test results.variables == res.variables
+    end
+end
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: You need to delete the results directory after the tests are done

File: test/test_chronology.jl [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+@testset "Testing to verify no gaps in the time_stamp" begin
+    for (ix, s) in enumerate(sim.stages)
+        results = load_simulation_results(stage[ix], sim_results)
+        resolution = convert(Dates.Minute, get_sim_resolution(s))
+        time_stamp = results.time_stamp
+        length = size(time_stamp,1)
+        test = results.time_stamp[1,1]:resolution:results.time_stamp[length,1]
+        @test time_stamp[!,:Range] == test
+    end
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: Here too, delete the `testing_reading_results` directory

File: test/runtests.jl [hunk 10]
```diff
@@ -39,7 +42,7 @@ if !Sys.iswindows()
         include("test_renewable_generation_constructors.jl")
         include("test_load_constructors.jl")
         include("test_storage_constructors.jl")
-        include("test_hydro_generation_constructors.jl")
+        #include("test_hydro_generation_constructors.jl")
```
Comment (sourabhdalvi) [inline test/runtests.jl]: You need to change this, hydro tests are passing now

File: test/test_chronology.jl [hunk 11]
```diff
@@ -0,0 +1,58 @@
+ipopt_optimizer = with_optimizer(Ipopt.Optimizer, print_level = 4)
+GLPK_optimizer = with_optimizer(GLPK.Optimizer, msg_lev = GLPK.MSG_ALL)
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: This is in `runtests.jl`

File: test/test_chronology.jl [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+base_dir = string(dirname(dirname(pathof(PowerSimulations))))
+DATA_DIR = joinpath(base_dir, "test/test_data")
+include(joinpath(DATA_DIR, "data_5bus_pu.jl"))
+include(joinpath(DATA_DIR, "data_14bus_pu.jl"))
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: This is in `get_test_data.jl`

File: test/test_chronology.jl [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+## ED Model Ref
+branches = Dict{Symbol, DeviceModel}()
+services = Dict{Symbol, PSI.ServiceModel}()
+devices = Dict{Symbol, DeviceModel}(:Generators => DeviceModel(PSY.ThermalStandard, PSI.ThermalDispatchNoMin, SemiContinuousFF(:P, :ON)),
+                                    :Ren => DeviceModel(PSY.RenewableDispatch, PSI.RenewableFullDispatch),
+                                    :Loads =>  DeviceModel(PSY.PowerLoad, PSI.StaticPowerLoad),
+                                    :ILoads =>  DeviceModel(PSY.InterruptibleLoad, PSI.DispatchablePowerLoad),
+                                    )       
+template_ed= OperationsTemplate(CopperPlatePowerModel, devices, branches, services);
+
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: This is in `test_simulation_models.jl`

File: test/test_chronology.jl [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+GLPK_optimizer = with_optimizer(GLPK.Optimizer)
```
Comment (sourabhdalvi) [inline test/test_chronology.jl]: Should be removed

File: test/test_load_simulation.jl [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+services = Dict{Symbol, PSI.ServiceModel}()
+devices = Dict{Symbol, DeviceModel}(:Generators => DeviceModel(PSY.ThermalStandard, PSI.ThermalDispatchNoMin, SemiContinuousFF(:P, :ON)),
+                                    :Ren => DeviceModel(PSY.RenewableDispatch, PSI.RenewableFullDispatch),
+                                    :Loads =>  DeviceModel(PSY.PowerLoad, PSI.StaticPowerLoad),
+                                    :ILoads =>  DeviceModel(PSY.InterruptibleLoad, PSI.DispatchablePowerLoad),
+                                    )       
+template_ed= OperationsTemplate(CopperPlatePowerModel, devices, branches, services);
+
+GLPK_optimizer = with_optimizer(GLPK.Optimizer)
+
```
Comment (sourabhdalvi) [inline test/test_load_simulation.jl]: Same applies to everything in this file, no need for redefinition

File: test/test_simulation_models.jl [hunk 16]
```diff
@@ -1,3 +1,25 @@
...
+branches = Dict{Symbol, DeviceModel}()
+services = Dict{Symbol, PSI.ServiceModel}()
+devices = Dict{Symbol, DeviceModel}(:Generators => DeviceModel(PSY.ThermalStandard, PSI.ThermalDispatchNoMin, SemiContinuousFF(:P, :ON)),
+                                    :Ren => DeviceModel(PSY.RenewableDispatch, PSI.RenewableFullDispatch),
+                                    :Loads =>  DeviceModel(PSY.PowerLoad, PSI.StaticPowerLoad),
+                                    :ILoads =>  DeviceModel(PSY.InterruptibleLoad, PSI.DispatchablePowerLoad),
+                                    )       
+template_ed= OperationsTemplate(CopperPlatePowerModel, devices, branches, services);
+
+GLPK_optimizer = with_optimizer(GLPK.Optimizer)
```
Comment (sourabhdalvi) [inline test/test_simulation_models.jl]: not sure if need this here, it already in `runtests.jl`

File: Project.toml [hunk 17]
```diff
@@ -22,6 +24,8 @@ Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 TimeSeries = "9e3dc215-6440-5c97-bce1-76c03772f85e"
 
 [compat]
+Colors = "~0.9"
+CSV = "=0.5.18"
```
Comment (jd-lara) [inline Project.toml]: change to `CSV = "=0.5.13`

File: src/PowerSimulations.jl [hunk 18]
```diff
@@ -89,24 +89,32 @@ export set_device_model!
 export set_branch_model!
 export set_device_model!
 ## Sim Model Exports
+export make_references
 export execute!
 ## Utils Exports
+export sim_results_container
```
Comment (jd-lara) [inline src/PowerSimulations.jl]: SimulationsResultsContainer if this is a Struct

**Pull request merged**